modules/core/src/main/scala/org/tessellation/infrastructure/dag/DAGService.scala

```diff
@@ -34,25 +35,31 @@ object DAGService {
           _.map(getBalance(_, address))
         }
 
-      private def getTotalSupply(snapshot: Signed[GlobalSnapshot]): (BigInt, SnapshotOrdinal) = {
+      private def calcTotalSupply(balances: Iterable[Balance], ordinal: SnapshotOrdinal): (BigInt, SnapshotOrdinal) = {
         val empty = BigInt(Balance.empty.coerce.value)
-        val supply = snapshot.value.info.balances.values
+        val supply = balances
           .foldLeft(empty) { (acc, b) =>
             acc + BigInt(b.coerce.value)
           }
-        val ordinal = snapshot.value.ordinal
 
         (supply, ordinal)
       }
 
       def getTotalSupply: F[Option[(BigInt, SnapshotOrdinal)]] =
         globalSnapshotStorage.head.map {
-          _.map(getTotalSupply)
+          _.map(s => calcTotalSupply(s.info.balances.values, s.ordinal))
+        }
+
+      def getFilteredOutTotalSupply: F[Option[(BigInt, SnapshotOrdinal)]] =
+        globalSnapshotStorage.head.map { maybeSignedSnapshot =>
+          maybeSignedSnapshot.map { snapshot =>
+            calcTotalSupply(snapshot.info.balances.filterNot { case (a, _) => lockedAddresses.contains(a) }.values, snapshot.ordinal)
```

Comment:
`filterNot` should take into account only the stardust address (or selected addresses), not all locked addresses (as it will change the total supply), and it will cause the total supply change in case of emergency when we would want to immediately lock some address in the future

Reply:
maybe it'd be wise to separate it into two lists: `lockedAddresses` / `filteredOutAddresses` 🤔 
or some additional flag next to locked address, an indicator to filter it our or no

Reply:
Since this is only rendered to user, I don't see a problem here. As long as the addresses are locked, they should not appear in total supply as they're unspendable (except for validation purposes from a separate function.)